core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1387,6 +1387,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Register an RDD to be persisted in memory and/or disk storage
    */
   private[spark] def persistRDD(rdd: RDD[_]) {
+    _executorAllocationManager.foreach { _ =>
+      logWarning(
+        s"Dynamic allocation currently does not support cached RDDs. Cached data for RDD " +
+        s"${rdd.id} will be deallocated when executors are removed.")
```

Comment:
nit: can we say "lost" instead of "deallocated"?  Deallocate has a connotation to me of purposely returned to a pool, where as this is more a situation of unpurposely returned to nowhere accessible.